test/testdrive/temporary.td

```diff
@@ -67,6 +67,19 @@ catalog item 'foo' already exists
 
 > CREATE OR REPLACE TEMPORARY MATERIALIZED VIEW foo AS SELECT * FROM v
 
+> CREATE TEMPORARY TABLE temp_t (a int, b text NOT NULL)
+
+! CREATE TEMP TABLE temp_t (a int, b text NOT NULL)
+catalog item 'temp_t' already exists
+
```

Comment:
In the presence of a non-temporary and a temporary table with the same name, ```DROP TABLE``` drops the non-temporary table:

```
pstoev=> create table t1 (f1 integer);
CREATE TABLE
pstoev=> create temporary table t1 (f1 integer);
CREATE TABLE
pstoev=> drop table t1; <- this seems to drop the non-temporary table
DROP TABLE
pstoev=> explain select * from t1;
      Optimized Plan      
--------------------------
 %0 =                    +
 | Get mz_temp.t1 (u2079)+ <- the temporary table remained
(1 row)
```

Reply:
I was able to fix this locally! (The same thing fixed this and the table masking issue you pointed out.)
```
materialize=> create table t1 (f1 integer);
CREATE TABLE
materialize=> create temporary table t1 (f1 integer);
CREATE TABLE
materialize=> drop table t1;
DROP TABLE
materialize=> explain select * from t1;
          Optimized Plan
----------------------------------
 %0 =                            +
 | Get materialize.public.t1 (u1)+

(1 row)
```